tests/event.test.ts

```diff
@@ -0,0 +1,248 @@
+      .createUsers([admin])
+      .write();
+
+    const eventChanges = {
+      title: 'new title',
+      description: 'new description',
+    };
+
+    const updateEventResponse = await ControllerFactory
+      .event(conn)
+      .updateEvent({ uuid: event.uuid }, { event: eventChanges }, admin);
+
+    expect(updateEventResponse.event).toMatchObject({ ...event.getPublicEvent(), ...eventChanges });
+  });
+
+  test('event attendance code cannot be updated to a duplicate one', async () => {
+    const conn = await DatabaseConnection.get();
+    const [event1] = EventFactory.with({ attendanceCode: 'code' });
+    const [event2] = EventFactory.with({ attendanceCode: 'another-code' });
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+
+    await new PortalState()
+      .createEvents([event1, event2])
+      .createUsers([admin])
+      .write();
+
+    event1.attendanceCode = event2.attendanceCode;
+
+    await expect(
+      ControllerFactory.event(conn).updateEvent({ uuid: event1.uuid }, { event: event1 }, admin),
+    ).rejects.toThrow('Attendance code has already been used');
+  });
+});
+
+describe('event covers', () => {
+  test('properly updates cover photo in database', async () => {
+    const conn = await DatabaseConnection.get();
+    const [event] = EventFactory.create(1);
+    const [admin] = UserFactory.with({ accessType: UserAccessType.ADMIN });
+    const cover = FileFactory.image(Config.file.MAX_EVENT_COVER_FILE_SIZE / 2);
+    const fileLocation = 'fake location';
+
+    await new PortalState()
+      .createUsers([admin])
+      .createEvents([event])
+      .write();
+
+    const eventCoverResponse = await ControllerFactory
+      .event(conn, null, Mocks.storage(fileLocation))
+      .updateEventCover(cover, { uuid: event.uuid }, admin);
+
+    expect(eventCoverResponse.event.cover).toEqual(fileLocation);
+  });
+
+  test('rejects upload if file size too large', async () => {
+    // TODO: implement once API wrappers exist (since multer validation can't be verified with function calls)
+  });
+});
+
+describe('event feedback', () => {
+  test('can be persisted and rewarded points when submitted for an event already attended', async () => {
+    const conn = await DatabaseConnection.get();
+    const event = EventFactory.fakeOngoingEvent();
+    const user = UserFactory.fake();
+    const feedback = FeedbackFactory.createEventFeedback(3);
+
+    await new PortalState()
+      .createUsers([user])
+      .createEvents([event])
+      .attendEvents([user], [event], false)
+      .write();
+
+    await ControllerFactory.event(conn).submitEventFeedback({ uuid: event.uuid }, { feedback }, user);
+
+    const attendanceResponse = await ControllerFactory.attendance(conn).getAttendancesForCurrentUser(user);
+
+    expect(attendanceResponse.attendances[0].feedback).toEqual(feedback);
+    expect(user.points).toEqual(event.pointValue + Config.pointReward.EVENT_FEEDBACK_POINT_REWARD);
```

Comment:
How is this passing? Shouldn't we need to refresh the `UserModel` before the points field is updated?

Reply:
https://github.com/acmucsd/membership-portal/blob/master/repositories/UserRepository.ts#L52

This line updates the points field directly on the `user` object, so no refresh is necessary.